src/main/java/com/titanic/airbnbclone/service/AccommodationService.java

```diff
@@ -3,132 +3,155 @@
-            Account foundAccount = accountRepository.findByEmail(userEmail);
-            List<ReservationInfoResponseDto> allData = new ArrayList<>();
-
-            // 현재 로그인된 사용자의 예약 내역을 확인해 DTO로 만드는 과정
-           for (Reservation reservation : foundAccount.getReservations()) {
-                Long accommodationId = reservation.getAccommodation().getId();
-                Accommodation foundAccommodation = accommodationRepository.findOneWithPictures(accommodationId)
-                        .orElseThrow(() -> new NoSuchEntityException(accommodationId));
-
-                ReservationInfoResponseDto reservationInfoResponseDto
-                        = ReservationInfoResponseDto.builder()
-                        .accommodationId(accommodationId)
-                        .hotelName(foundAccommodation.getName())
-                        .reservation(AccountReservationResponseDto.of(reservation))
-                        .urls(foundAccommodation.getPictures())
-                        .build();
-                allData.add(reservationInfoResponseDto);
-            }
-
-            return ReservationInfoResponseDtoList.builder()
-                    .status(StatusEnum.SUCCESS.getStatusCode())
-                    .allData(allData)
-                    .build();
-        } catch (Exception e) {
-            return ReservationInfoResponseDtoList.builder()
-                    .status(StatusEnum.ACCEPTED.getStatusCode())
-                    .build();
-        }
+        // 1차 필터링 조건(지역, 예약인원)으로 찾은 숙박들
+        List<Accommodation> filteredAccommodation = accountRepository.filterAccommodation(filterRequestDto);
+
+        // 1차 필터링된 숙박 중 예약있는 숙박
+        List<Accommodation> reservedAccommodations = filteredAccommodation .stream()
+                .filter(Accommodation::hasReservation)
+                .collect(Collectors.toList());
+
+        // 예약이 있는 숙박 중에서 요청된 날짜로 예약이 불가능한 숙박들
+        List<Accommodation> cannotReserved = reservedAccommodations.stream()
+                .filter(accommodation -> !accommodation.isReservable(filterRequestDto.getStartDate(), filterRequestDto.getEndDate()))
+                .collect(Collectors.toList());
+
+        // 전체 숙박 중 예약 불가능한 숙박 제거하는 과정
+        cannotReserved.stream()
+                .map(filteredAccommodation::remove)
+                .collect(Collectors.toList());
+
+        List<AccommodationResponseDto> allData = filteredAccommodation.stream()
+                .map(AccommodationResponseDto::of)
+                .collect(Collectors.toList());
+
+        return AccommodationResponseDtoList.builder()
+                .status(StatusEnum.SUCCESS.getStatusCode())
+                .allData(allData)
+                .build();
+    }
+
+    // HttpServletRequest에서 유저이메일 꺼내오는 메서드
+    private String getAccountEmail(HttpServletRequest request) {
+        return (String) request.getAttribute(OauthEnum.USER_EMAIL.getValue());
+    }
```

Comment:
getAccountEmail()했을 때 AccountRepository 로직까지 한번에 처리하면 findAccount할 때 이중괄호를 없애 좀 더 가독성이 좋아요.
```suggestion
    private Account getAccount(HttpServletRequest request) {
        String email = (String) request.getAttribute(OauthEnum.USER_EMAIL.getValue());
        return accountRepository.findByEmail(email);
    }
```

Reply:
굿굿 좋네요!!